app/src/main/java/cm/aptoide/pt/search/SearchManager.java

```diff
@@ -52,19 +63,88 @@ public SearchManager(SharedPreferences sharedPreferences, TokenInvalidator token
     this.moPubAdsManager = moPubAdsManager;
     this.appBundlesVisibilityManager = appBundlesVisibilityManager;
     this.searchRepository = searchRepository;
+    this.downloadStatusManager = downloadStatusManager;
+    this.appCenter = appCenter;
   }
 
   public Observable<SearchAdResult> getAdsForQuery(String query) {
     return adsRepository.getAdsFromSearch(query)
         .map(minimalAd -> new SearchAdResult(minimalAd));
   }
 
-  public Single<SearchResult> searchAppInStores(String query, List<Filter> filters) {
+  public Completable searchAppInStores(String query, List<Filter> filters) {
     return accountManager.hasMatureContentEnabled()
         .first()
         .toSingle()
-        .flatMap(matureEnabled -> searchRepository.generalSearch(query, getSearchFilters(filters),
-            matureEnabled));
+        .flatMapCompletable(
+            matureEnabled -> searchRepository.generalSearch(query, getSearchFilters(filters),
+                matureEnabled));
+  }
+
+  public Observable<SearchResult> observeSearchResults() {
+    return searchRepository.observeSearchResults()
+        .switchMap(result -> {
+          List<SearchAppResult> list = result.getSearchResultsList();
+          if (!list.isEmpty() && list.get(0)
+              .isHighlightedResult()) {
+            if (result.isFreshResult()) {
+              return Observable.mergeDelayError(Observable.just(result),
+                  observeHighlightedSearchResult(result));
+            }
+            return observeHighlightedSearchResult(result);
+          }
+          return Observable.just(result);
+        });
+  }
+
+  public Observable<SearchResult> observeHighlightedSearchResult(SearchResult result) {
+    SearchAppResult first = result.getSearchResultsList()
+        .get(0);
+    return Observable.combineLatest(getHighlightedSearchResult(result),
+        downloadStatusManager.loadDownloadModel(first.getMd5(), first.getPackageName(),
+            first.getVersionCode(), null, first.getStoreId(),
+            first.hasAdvertising() || first.hasBilling()),
+        loadAppScreenShots(first.getAppId(), first.getStoreName(), first.getPackageName()),
+        (r, downloadModel, screenshots) -> mergeSearchResult(r, downloadModel, screenshots));
+  }
+
+  private Observable<List<AppScreenshot>> loadAppScreenShots(long appId, String storeName,
+      String packageName) {
+    return Observable.mergeDelayError(Observable.just(null),
+        appCenter.unsafeLoadDetailedApp(appId, storeName, packageName)
```

Comment:
As we've discussed, it is because in this method there is no mechanism to avoid doing two requests at the same time. This is important to avoid issues with rapidly changing filters.